### ISSUE TITLE
Create prepublish script

### ISSUE THREAD

**Issue by chitchu** [ractive-foundation/ractive-foundation#373]
The current workflow for publishing new components are the following:

1. run `gulp dist`
1. commit `dist` folder to the repo
1. bump tags
1. CI publishes to npm

Binaries should not be committed to the repo as this creates meaningless commits.

Suggestions:

1. untrack `dist` folder (add to `.gitignore`)
1. create a `prepublish` script in `package.json` that would run `gulp dist`

nice to have: create a `.npmignore` file and add all uncompiled source code. when users run `npm i ractive-foundation` it should only install compiled files/binaries. 

currently this is what ractive-foundation looks like inside the `node_modules`

```sh
./node_modules/ractive-foundation/
├── README.md
├── TODO.md
├── dist
├── gulpfile.js
├── package.json
├── scripts
├── src
└── tasks

4 directories, 4 files

```

**Comment by nathf** [comment]
All you have to run is `npm version <major|minor|patch>` and it will run that process (run dist, bump, commit) for you, but agree the process could be improved to not include the generated assets, can't remember the reason behind that.

Edit: Some super brief docs here: https://github.com/ractive-foundation/ractive-foundation/wiki/Releasing-RF-to-npm

**Comment by chitchu** [comment]
`npm version <major|minor|patch>` this works well. But for PR's someone needs to run this command manually.

**Comment by nathf** [comment]
Yep totally agree - not ideal. How were you thinking of changing the publishing process? Through GH releases or?

**Comment by chitchu** [comment]
Actually now that I think about it, tagging should only be done manually. But let CI publish to npm on tag changes. I guess the biggest change that needs to be done it adding the `dist` folder to untracked files and use `prepublish` to build the dist files.

So basically:

1. new component PR gets merged to `master`
1. someone bumps semver
1. CI detects a tagged release and runs `npm publish`
1. `prepublish` kicks in and creates the `dist` folder
1. CI publishes the repo along with the created `dist` folder
1. ???
1. Profit!

**Comment by nathf** [comment]
I think GH releases make sense for this change - and deprecate `npm version` releasing, it will also open up the opportunity to provide change logs per release. Looks like a sound plan!
